### PR TITLE
update glfw3 to 3.1.2

### PR DESCRIPTION
**Pull request by andrewrk**

### CODE REVIEW

File: src/glfw3.mk [hunk 1]
```diff
@@ -3,19 +3,18 @@
...
 PKG             := glfw3
 $(PKG)_IGNORE   :=
-$(PKG)_VERSION  := 3.1
-$(PKG)_CHECKSUM := 4948d5091d71a2249dc6d7e3effab2066089334844cea5cef0489b4a575b0bce
+$(PKG)_VERSION  := 3.1.2
+$(PKG)_CHECKSUM := 6ac642087682aaf7f8397761a41a99042b2c656498217a1c63ba9706d1eef122
 $(PKG)_SUBDIR   := glfw-$($(PKG)_VERSION)
-$(PKG)_FILE     := glfw-$($(PKG)_VERSION).tar.gz
-$(PKG)_URL      := http://$(SOURCEFORGE_MIRROR)/project/glfw/glfw/$($(PKG)_VERSION)/$($(PKG)_FILE)
+$(PKG)_SUBDIR   := glfw-$($(PKG)_VERSION)
```
Comment (TimothyGu) [inline src/glfw3.mk]: Why the duplicated SUBDIR?

File: src/glfw3.mk [hunk 2]
```diff
@@ -3,19 +3,18 @@
...
 
 define $(PKG)_UPDATE
-    $(WGET) -q -O- 'http://sourceforge.net/projects/glfw/files/glfw/' | \
-    $(SED) -n 's,.*/\([0-9][^"]*\)/".*,\1,p' | \
-    grep '^3\.' | \
-    $(SORT) -V | \
-    tail -1
+    $(WGET) -q -O- 'https://github.com/glfw/glfw/releases' | \
+    $(SED) -n 's,.*/archive/\([0-9][^>]*\)\.tar.*,\1,p' | \
+    head -1
```
Comment (TimothyGu) [inline src/glfw3.mk]: https://github.com/mxe/mxe/blob/master/Makefile#L146

File: src/glfw3.mk (hunk 1 above)
Comment (andrewrk) [inline src/glfw3.mk]: A mistake.

**Comment by andrewrk** [comment]
Thanks for the review. I pushed attempt number 2.

**Comment by TimothyGu** [comment]
LGTM.

**Pull request merged**